app/models/constants.ts

```diff
@@ -32,3 +32,114 @@ export const NATIONAL_AVERAGES: nationalAverageType = {
 };
 
 export const MAINTENANCE_LEVELS = [0.015, 0.02, 0.0375] as const;
+
+/** Type for storing component values and depreciation*/
+export type componentBreakdownType = {
+  /** Component value as percentage of total house value */
+  percentageOfHouse: number, 
+  /** Percentage of the component's total value that is written-down yearly */
+  depreciationPercentageYearly: number,
+  /** Percentage of the yearly maintenance spend allocated to component */
+  percentOfMaintenanceYearly: number
+}
+
+export type houseBreakdownType = {
+  foundations: componentBreakdownType,
+  structureEnvelope: componentBreakdownType,
+  cladding: componentBreakdownType,
+  roofing: componentBreakdownType,
+  windows: componentBreakdownType,
+  internalLinings: componentBreakdownType,
+  bathroomFixtures: componentBreakdownType,
+  fitout: componentBreakdownType,
+  kitchenUnits: componentBreakdownType,
+  electricalAppliances: componentBreakdownType,
+  electricalServices: componentBreakdownType,
+  ventilationServices: componentBreakdownType,
+  waterAndHeatingServices: componentBreakdownType,
+  floorCoverings: componentBreakdownType,
+  landscaping: componentBreakdownType,
+}
+
+/** 
+ * Object (key = component, value = another object with `percentageOfHouse`, `deprecationPercentageYearly` and `percentOfMaintenanceYearly`). Contains a breakdown of a house by component, and the percentage of total value it constitutes:
+ * foundations 21%, structure 25%, cladding 4%, roofing 4%, windows 4%, internal linings 4%, bathroom 4%, fitout 5%,
+ * kitchen units 4%, electrical appliances 4%, electrical services 4%, ventilation services 4%, water and heating services 8%,
+ * floor coverings 2%, landscaping 3%*/
```

Comment:
It might be worth encoding these as variables also?

Reply:
Thanks for the keen eyes!

Do you mean encoding as variables in addition to storing them inside the objects? For instance, instead of:
``` 
export const HOUSE_BREAKDOWN_PERCENTAGES: houseBreakdownType = {
  foundations: {
    percentageOfHouse: .21,
    depreciationPercentageYearly: 0,
    percentOfMaintenanceYearly: 0
  },
  ```
  We might have something like
  ``` 
const foundationsPercent = .21
  export const HOUSE_BREAKDOWN_PERCENTAGES: houseBreakdownType = {
  foundations: {
    percentageOfHouse: foundationsPercent,
...
```
?